bundles/io/org.eclipse.smarthome.io.transport.mdns/src/main/java/org/eclipse/smarthome/io/transport/mdns/internal/MDNSClientImpl.java

```diff
@@ -226,4 +261,29 @@ public void onChanged(List<CidrAddress> added, List<CidrAddress> removed) {
             createJmDNSByAddress(address);
         }
     }
+
+    private void getConfigParameters(Map<String, Object> parameters) {
+        useOnlyOneAddress = getConfigParameter(parameters, CONFIG_USER_ONLY_ONE_ADDRESS, false);
+        ignoreIPv6 = getConfigParameter(parameters, CONFIG_IGNORE_IPV6, false);
+    }
+
+    private boolean getConfigParameter(Map<String, Object> parameters, String parameter, boolean defaultValue) {
+        if (parameters == null) {
+            return defaultValue;
+        }
+        Object value = parameters.get(parameter);
+        if (value == null) {
+            return defaultValue;
+        }
+        if (value instanceof Boolean) {
+            return (Boolean) value;
+        }
+        if (value instanceof String) {
+            return Boolean.valueOf((String) value);
+        } else {
+            logger.warn("ignoring invalid type {} for parameter {}", value.getClass().getName(), parameter);
```

Comment:
I would note in the warning that the default value will be used.

Reply:
I will add that